### DISCUSSION TITLE
Can't switch chains on mobile browser with MetaMask

### DISCUSSION THREAD

**Discussion by Markcode1992** [reown-com/appkit#2705] [General]
Does anyone else have issues with chain switching? When I use a mobile browser and Metamask I cannot switch chains. Sometimes the first time works and sometimes it doesn't. Also often when it does work I get an error: error: missing or invalid. emit() chainId: eip155:56

I have tried multiple webapps using this wagmi and it happened on all of them.

**Comment by rtomas** [comment]
do you have some dApp to test ?
Could you check if it works properly with our test lab at https://lab.web3modal.com/?
